libs/datamodel/core/src/configuration/source/mod.rs

```diff
@@ -1,14 +1,13 @@
+mod builtin_datasource_providers;
+mod datasource;
+mod datasource_provider;
 mod loader;
 mod serializer;
-mod traits;
-
-pub mod builtin;
 
 #[cfg(feature = "mssql")]
 pub use builtin::MSSQL_SOURCE_NAME;
-// TODO: i think these constants should move to a more central place.
-pub use builtin::{MYSQL_SOURCE_NAME, POSTGRES_SOURCE_NAME, SQLITE_SOURCE_NAME};
-//pub use json::{render_sources_to_json, render_sources_to_json_value, sources_from_json_value_with_plugins};
+pub use builtin_datasource_providers::*;
+pub use datasource::*;
+pub use datasource_provider::*;
```

Comment:
One of the things I've struggled with when onboarding is tracking down where variables come from. It makes it hard to understand how the separate modules relate to each other and create a mental model of the crates. I'm not a fan of switching from explicit imports to globs here.

Reply:
Why do you think this is a switch? I kept the same pattern for reexport that was present before as well.

Reply:
![Screenshot_2020-06-30 Implement Dynamic Datasource Providers by mavilein · Pull Request #800 · prisma prisma-engines](https://user-images.githubusercontent.com/2467194/86118165-955bb280-bad0-11ea-9f21-68a41afaa579.png)

> Why do you think this is a switch? 

This reads as named imports being replaced with globs.

---

> I kept the same pattern for reexport that was present before as well.

I know? Like I mentioned earlier in this thread: this is one of the patterns that makes the codebase hard to navigate. To me it comes off as convenient to author, but it trades away legibility. We don't have to change all instances of this pattern in a single go, but as a baseline I would like us to not introduce *more* of these and think it's worth bringing up in review.